cpp/include/cudf/detail/copy_if_else.cuh

```diff
@@ -71,23 +64,17 @@ __launch_bounds__(block_size) __global__
   size_type warp_cur = warp_begin + warp_id;
   size_type index    = tid;
   while (warp_cur <= warp_end) {
-    bool in_range = (index >= begin && index < end);
-
-    bool valid = true;
-    if (has_validity) {
-      valid = in_range && (filter(index) ? thrust::get<1>(lhs[index]) : thrust::get<1>(rhs[index]));
-    }
-
-    // do the copy if-else
-    if (in_range) {
-      out.element<T>(index) = filter(index) ? static_cast<T>(thrust::get<0>(lhs[index]))
-                                            : static_cast<T>(thrust::get<0>(rhs[index]));
+    bool valid = false;
+    if (index >= begin && index < end) {
+      auto value = filter(index) ? lhs[index] : rhs[index];
+      valid      = !has_validity || value.has_value();
+      if (valid) { out.element<T>(index) = static_cast<T>(value.value()); }
     }
 
     // update validity
     if (has_validity) {
       // the final validity mask for this warp
-      int warp_mask = __ballot_sync(0xFFFF'FFFF, valid && in_range);
+      int warp_mask = __ballot_sync(0xFFFF'FFFF, valid);
```

Comment:
```suggestion
      int warp_mask = __ballot_sync(0xFFFF'FFFF, value.has_value());
```

Reply:
The `value` is not available in this scope.